From 4f030533d171da6f84d3be4881620a02c1e02a6d Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 7 Mar 2025 10:05:45 +0100
Subject: [PATCH 1630/1695] drivers: stm32_rif: add mask to get static CID of a
 peripheral

This mask can be use to check the peripheral RIF assignment.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 560ffbafedf40955e4194a3d23ee2a0ce040e04e drivers: stm32_rif: add mask to get static CID of a peripheral]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Id0f1aba25790a4c9bd082f060c357919a6040003
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/442714
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/include/drivers/stm32_rif.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/include/drivers/stm32_rif.h b/core/include/drivers/stm32_rif.h
index 521824059..5eebf6e6b 100644
--- a/core/include/drivers/stm32_rif.h
+++ b/core/include/drivers/stm32_rif.h
@@ -16,6 +16,7 @@
  */
 #define _CIDCFGR_CFEN			BIT(0)
 #define _CIDCFGR_SCID_SHIFT		U(4)
+#define _CIDCRGR_SCID_MASK		GENMASK_32(6, 4)
 #define _CIDCFGR_SEMEN			BIT(1)
 #define _CIDCFGR_SEMWL_SHIFT		U(16)
 #define _CIDCFGR_SEMWL(x)		BIT(_CIDCFGR_SEMWL_SHIFT + (x))
-- 
2.47.2

